### PR TITLE
Use Stream Service Functions

### PR DESCRIPTION
**Pull request by brettephillips**
- Updated code to use built in stream layer methods rather than add and removing the stream service every time. I have confirmed that the data is being sent to the backend and messages received are altered:

     Screen Shot 2022-12-21 at 1 20 43 PM

Including UI changes as well:
- Updated UI to use the new calcite-components library
     - Added toaster to show messages (errors, successes, warnings) 
- Added basemap switcher

### CODE REVIEW

**Review by hanoch — APPROVED** [review]

**Review by hanoch — COMMENTED** [review]
A few nit-pick comments:

1. The `Switch Basemap` button doesn't seem to be semi-transparent.
2. The textbox alignments are weird, see the screenshot below.
3. Is there a way to scroll through all toaster messages without needing to delete/dismiss a message at a time? 
4. Let's split the `Spatial Filter` button into two buttons - `Set Spatial Filter` and `Clear Spatial Filter` (that we can disable/enable when there is no spatial filter defined). This will allow us to update the spatial filter on the fly without needing to clear it first.

image

File: index.html [hunk 1]
```diff
@@ -5,17 +5,23 @@
...
     <link
       rel="stylesheet"
-      href="https://js.arcgis.com/4.14/esri/themes/light/main.css"
+      href="https://js.arcgis.com/3.42/esri/css/esri.css"
     />
     <link
       rel="stylesheet"
-      href="https://js.arcgis.com/3.42/esri/css/esri.css"
+      type="text/css"
+      href="https://js.arcgis.com/calcite-components/1.0.0-beta.97/calcite.css"
```
Comment (hanoch) [inline index.html:13]: Is there a non-beta calcite release already?

Comment (brettephillips) [reply index.html:13]: There is no non-beta release yet! The calcite-components repository is relatively new and there has been a big push lately to move toward it.

**Comment by brettephillips** [comment]
> A few nit-pick comments:
> 
> 1. The `Switch Basemap` button doesn't seem to be semi-transparent.
> 2. The textbox alignments are weird, see the screenshot below.
> 3. Is there a way to scroll through all toaster messages without needing to delete/dismiss a message at a time?
> 4. Let's split the `Spatial Filter` button into two buttons - `Set Spatial Filter` and `Clear Spatial Filter` (that we can disable/enable when there is no spatial filter defined). This will allow us to update the spatial filter on the fly without needing to clear it first.
> 
> image

@hanoch - For items 1 and 2, please try and clear your cache. I am seeing this show up correctly on my end. If it still shows weird on your end, we can troubleshoot together. With 3, I will look into the calcite library to see if this is possible or at least see if we can stack alerts. In addition, I will also handle number 4 as requested.

**Comment by hanoch** [comment]
@brettephillips I can see that items 1,2, and 4 are now addressed:
image

Looks good to merge.

**Review by hanoch — APPROVED** [review]
Looks good to merge.

File: streamServiceViewerStyles.css [hunk 2]
```diff
@@ -186,6 +186,16 @@ section:first-of-type .config-header {
...
 }
 
+.input-row {
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+}
+
+.prev-element {
+	margin-right: 5px;
```
Comment (hanoch) [inline streamServiceViewerStyles.css:196]: Do we need to adjust the spaces/tabs of this line?

Comment (brettephillips) [reply streamServiceViewerStyles.css:196]: Yes, I will run the formatter on this file

**Pull request merged**